arm9/src/unk_020851B8.c

```diff
@@ -0,0 +1,61 @@
+#include "global.h"
+#include "unk_020851B8.h"
+#include "constants/items.h"
+#include "heap.h"
+
+static inline void _clear(u8 *ptr)
+{
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+    *ptr++ = 0;
+}
+
+THUMB_FUNC struct UnkStruct_020851B8 *FUN_020851B8(u32 heap_id)
+{
+    struct UnkStruct_020851B8 *ptr = AllocFromHeap(heap_id, sizeof(struct UnkStruct_020851B8));
+    _clear((u8 *)ptr);
+    return ptr;
+}
```

Comment:
Upon further investigation, this also matches with the metrowerks `__builtin__clear` macro:

```c
THUMB_FUNC struct UnkStruct_020851B8 *FUN_020851B8(u32 heap_id)
{
    struct UnkStruct_020851B8 *ptr = AllocFromHeap(heap_id, sizeof(struct UnkStruct_020851B8));
    __builtin__clear(ptr, sizeof(struct UnkStruct_020851B8));
    return ptr;
}
```